Userland/Services/WebDriver/Session.cpp

```diff
@@ -106,4 +114,55 @@ Web::WebDriver::Response Session::stop()
     return JsonValue {};
 }
 
+// 11.4 Get Window Handles, https://w3c.github.io/webdriver/#dfn-get-window-handles
+Web::WebDriver::Response Session::get_window_handles() const
+{
+    // 1. Let handles be a JSON List.
+    JsonArray handles {};
+
+    // 2. For each top-level browsing context in the remote end, push the associated window handle onto handles.
+    for (auto const& window_handle : m_windows.keys()) {
+        handles.append(JsonValue(window_handle));
+    }
+
+    // 3. Return success with data handles.
+    return JsonValue { move(handles) };
+}
+
+Web::WebDriver::Response Session::switch_to_window(StringView const& handle)
+{
+    // 4. If handle is equal to the associated window handle for some top-level browsing context in the
+    //    current session, let context be the that browsing context, and set the current top-level
+    //    browsing context with context.
+    //    Otherwise, return error with error code no such window.
+    auto const& maybe_window = m_windows.get(handle);
+
+    if (!maybe_window.has_value())
+        return Web::WebDriver::Error::from_code(Web::WebDriver::ErrorCode::NoSuchWindow, "Window not found");
+
+    m_current_window_handle = String::from_utf8(handle).release_value_but_fixme_should_propagate_errors();
+
+    // FIXME: 5. Update any implementation-specific state that would result from the user selecting the current
+    //          browsing context for interaction, without altering OS-level focus.
+
+    // 6. Return success with data null.
+    return JsonValue {};
+}
+
+Web::WebDriver::Response Session::close_window()
```

Comment:
Spec link here:
```
// 11.3 Switch to Window, https://w3c.github.io/webdriver/#dfn-switch-to-window
```

Also please keep these function definitions (and their declarations) in spec order. They're defined in spec order everywhere else, I'd like to keep it that way :)

Reply:
yes, makes sense, I added spec links and rearranged them in definition order.